src/topology/separation.lean

```diff
@@ -654,6 +654,28 @@ begin
   exact (set.to_finite _).is_closed
 end
 
+lemma preconnected_space.trivial_of_discrete [preconnected_space α] [discrete_topology α] :
+  subsingleton α :=
+begin
+  rw ←not_nontrivial_iff_subsingleton,
+  rintro ⟨x, y, hxy⟩,
+  obtain h | h := is_clopen_iff.mp (is_clopen_discrete ({y} : set α)),
```

Comment:
You can use [eq_univ_of_nonempty_clopen](https://leanprover-community.github.io/mathlib_docs/topology/connected.html#eq_univ_of_nonempty_clopen)

Reply:
Otherwise LGTM. Thanks!
bors d+

Reply:
thanks, I've done that. I've also renamed it to `is_clopen.eq_univ` for dot notation; I did this in a separate commit so it's easy to undo if you don't think it's a good idea. `is_clopen.eq_univ_of_nonempty` also seems okay, or even `set.nonempty.clopen_eq_univ`; I just think it'd be nice to have some dot notation here.